src/model/test/CoilCoolingDXSingleSpeedThermalStorage_GTest.cpp

```diff
@@ -848,12 +859,103 @@ TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_HeatCoolFuelTypes) {
   testAppGFuelTypeEquality({}, coilCoolingDXSingleSpeedThermalStorage.appGHeatingFuelTypes());
 }
 
-TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_containingHVACComponent) {}
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_containingHVACComponent) {
+  {  // AirLoopHVACUnitarySystem
+    Model model;
 
-TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_containingZoneHVACComponent) {}
+    CoilCoolingDXSingleSpeedThermalStorage dx(model);
+    EXPECT_TRUE(dx.isRemovable());
 
-TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_addToNode) {}
+    AirLoopHVACUnitarySystem unitary(model);
+    EXPECT_TRUE(unitary.setCoolingCoil(dx));
+    ASSERT_TRUE(unitary.coolingCoil());
+    EXPECT_EQ(dx, unitary.coolingCoil().get());
 
-TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_clone) {}
+    ASSERT_TRUE(dx.containingHVACComponent());
+    EXPECT_EQ(unitary, dx.containingHVACComponent().get());
 
-TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_cloneParent) {}
+    // Shouldn't be able to remove DX since it's contained in a HVAC Component
+    EXPECT_FALSE(dx.isRemovable());
+    auto rmed = dx.remove();
+    EXPECT_EQ(0u, rmed.size());
+    ASSERT_TRUE(unitary.coolingCoil());
+    EXPECT_EQ(dx, unitary.coolingCoil().get());
+  }
```

Comment:
maybe test you can setCoolingCoil with another type of Unitary sys then?

Reply:
Same: test with the ZoneHVACComponents.

Reply:
setCoolingCoil for these system types should throw? Do we have examples/precedents for doing this?